shell/platform/android/hardware_buffer_external_texture_vk.cc

```diff
@@ -0,0 +1,60 @@
+
+#include "flutter/shell/platform/android/hardware_buffer_external_texture_vk.h"
+
+#include "flutter/impeller/renderer/backend/vulkan/android_hardware_buffer_texture_source_vk.h"
+#include "flutter/impeller/renderer/backend/vulkan/texture_vk.h"
+#include "flutter/shell/platform/android/ndk_helpers.h"
+#include "impeller/core/formats.h"
+#include "impeller/core/texture_descriptor.h"
+#include "impeller/display_list/dl_image_impeller.h"
+
+namespace flutter {
+
+HardwareBufferExternalTextureVK::HardwareBufferExternalTextureVK(
+    const std::shared_ptr<impeller::ContextVK>& impeller_context,
+    int64_t id,
+    const fml::jni::ScopedJavaGlobalRef<jobject>& image_texture_entry,
+    const std::shared_ptr<PlatformViewAndroidJNI>& jni_facade)
+    : HardwareBufferExternalTexture(id, image_texture_entry, jni_facade),
+      impeller_context_(impeller_context) {}
+
+HardwareBufferExternalTextureVK::~HardwareBufferExternalTextureVK() {}
+
+void HardwareBufferExternalTextureVK::ProcessFrame(PaintContext& context,
+                                                   const SkRect& bounds) {
+  if (state_ == AttachmentState::kUninitialized) {
+    // First processed frame we are attached.
+    state_ = AttachmentState::kAttached;
+  }
+
+  AHardwareBuffer* latest_hardware_buffer = GetLatestHardwareBuffer();
+  if (latest_hardware_buffer == nullptr) {
+    FML_LOG(WARNING) << "GetLatestHardwareBuffer returned null.";
+    return;
+  }
+
+  AHardwareBuffer_Desc hb_desc = {};
+  flutter::NDKHelpers::AHardwareBuffer_describe(latest_hardware_buffer,
+                                                &hb_desc);
+
+  impeller::TextureDescriptor desc;
+  desc.storage_mode = impeller::StorageMode::kDevicePrivate;
+  desc.size = {static_cast<int>(bounds.width()),
+               static_cast<int>(bounds.height())};
+  desc.format = impeller::PixelFormat::kR8G8B8A8UNormInt;
```

Comment:
Do we need to worry about checking the pixel format at runtime?

Reply:
Yes we do, I'm going to leave this as a TODO for now though.

Reply:
👍